### PR TITLE
Upgrade dfx, base library and CI scripts.

### PR DESCRIPTION
**Pull request by matthewhammer**

### CODE REVIEW

**Comment by matthewhammer** [comment]
My best guess is that something is going wrong with serialization of the result (?)

It's hard to say without diving deeper into the compiled Wasm, or its behavior, I fear.

**Comment by matthewhammer** [comment]
@chenyan-dfinity There's something new happening with the latest `moc` and `dfx`.

Compiling and installing is fine, but during the more complex test, the compiled Wasm seems to hang right when it is supposed to return the (complex) result rendering.

Strangely, my other projects that use this same serialized type does not seems to have this issue -- e.g., [this Alice-Bob chat example](https://github.com/matthewhammer/MirrorGarden/blob/main/scripts/MirrorGarden-start-replica.sh) still works with the dfx version I am trying to use here (0.6.16).

What should I do to debug this further, I wonder?

File: test/Calc.mo [hunk 1]
```diff
@@ -24,12 +34,26 @@ actor {
...
+    Debug.print "calc.eval(exp) done";
+
     calc.engine.draw().logEventLast();
-    calc.engine.draw().getResult()
+
+    Debug.print "calc.engine.draw().logEventLast() done";
+
+    let r = calc.engine.draw().getResult();
+
+    Debug.print "calc.engine.draw().getResult() done";
```
Comment (matthewhammer) [inline test/Calc.mo:55]: @chenyan-dfinity The test on my laptop and on CI do the same thing: They get past this line, but no further.  The call never returns, and the replica says something strange about buffer/queue sizes.

Comment (chenyan-dfinity) [reply test/Calc.mo:55]: Can you try `dfx canister call --output raw`? I suspect it's the pretty printer that is too slow.

Comment (matthewhammer) [reply test/Calc.mo:55]: Oh, I see
Right.

Comment (matthewhammer) [reply test/Calc.mo:55]: Will try now....Thanks!

Comment (matthewhammer) [reply test/Calc.mo:55]: Bingo -- That was it.

And now that you remind me, I recall having this same issue months back, probably for the same datatype.  : )

I'll make a sticky note this time : ). Thanks again for the help!

Comment (chenyan-dfinity) [reply test/Calc.mo:55]: You can file an issue at candid repo. You are the only one who has large output so far :)

**Comment by matthewhammer** [comment]
Memo to future self: Remember Yan's advice: https://github.com/matthewhammer/motoko-adapton/pull/16#discussion_r551540736

(try doing `dfx canister call --output raw` to avoid the slow pretty-printer)

**Pull request merged**